tests/Resources/ti.ui.view.test.js

```diff
@@ -23,18 +23,12 @@ describe('Titanium.UI.View', function () {
 
 	before(finish => {
 		rootWindow = Ti.UI.createWindow({ exitOnClose: false });
-		rootWindow.addEventListener('open', () => finish());
-		rootWindow.open();
+		rootWindow.open().then(() => finish()).catch(e => finish(e));
 	});
 
 	function closeWindow(win, done) {
 		if (win && !win.closed) {
-			win.addEventListener('close', function listener () {
-				win.removeEventListener('close', listener);
-				win = null;
-				done();
-			});
-			win.close();
+			win.close().then(() => done()).catch(_e => done());
```

Comment:
<!--
  0 failure: 
  2 warning:  tests/Resources/t..., tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.view.test.js;
  Line: 31;
-->

- :warning: tests/Resources/ti.ui.view.test.js line 31 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)
- :warning: tests/Resources/ti.ui.view.test.js line 31 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)